src/Neo/SmartContract/Native/Notary.cs

```diff
@@ -0,0 +1,342 @@
+// Copyright (C) 2015-2025 The Neo Project.
+//
+// Notary.cs file belongs to the neo project and is free
+// software distributed under the MIT software license, see the
+// accompanying file LICENSE in the main directory of the
+// repository or http://www.opensource.org/licenses/mit-license.php
+// for more details.
+//
+// Redistribution and use in source and binary forms with or without
+// modifications are permitted.
+
+#nullable enable
+#pragma warning disable IDE0051
+
+using Neo.Cryptography;
+using Neo.Cryptography.ECC;
+using Neo.Extensions;
+using Neo.Network.P2P;
+using Neo.Network.P2P.Payloads;
+using Neo.Persistence;
+using Neo.SmartContract.Manifest;
+using Neo.VM;
+using Neo.VM.Types;
+using System;
+using System.Linq;
+using System.Numerics;
+using Array = Neo.VM.Types.Array;
+
+namespace Neo.SmartContract.Native
+{
+    /// <summary>
+    /// The Notary native contract used for multisignature transactions forming assistance.
+    /// </summary>
+    public sealed class Notary : NativeContract
+    {
+        /// <summary>
+        /// A default value for maximum allowed NotValidBeforeDelta. It is set to be
+        /// 20 rounds for 7 validators, a little more than half an hour for 15-seconds blocks.
+        /// </summary>
+        private const int DefaultMaxNotValidBeforeDelta = 140;
```

Comment:
BeforeDeta?

Maybe this variable can be renamed, when start reviewing right on here I do not get its functionalities.

Reply:
> BeforeDeta?

`MaxNotValidBeforeDelta` is a maximum value of delta between the current height and NotValidBefore attribute height. `DefaultMaxNotValidBeforeDelta` is the default for this value.

I don't think it should be renamed. It reflects the meaning of this variable.

I will add Notary related documentation in a separate PR, once this PR is merged.

Reply:
> I will add Notary related documentation in a separate PR, once this PR is merged.

Is this documentation already there somewhere, or in planning? If not, could you provide an ETA?